src/ansi-c/cprover_builtin_headers.h

```diff
@@ -12,6 +12,9 @@ __CPROVER_bool __CPROVER_is_invalid_pointer(const void *);
 _Bool __CPROVER_is_zero_string(const void *);
 // a singly-linked null-terminated dynamically-allocated list
 __CPROVER_bool __CPROVER_is_list();
+__CPROVER_bool __CPROVER_is_dll();
+__CPROVER_bool __CPROVER_is_cyclic_dll();
+__CPROVER_bool __CPROVER_is_sentinel_dll();
```

Comment:
Is this really usable by anyone as the type-checking code is the only place that documents how many arguments those take, and what types those ought to take?!

Reply:
I think to make these really usable we need to have an extension mechanism for inductive data types. There are really very many ways of defining datatypes in C, and each code base will differ slightly.